system/jhtree/jhtree.cpp

```diff
@@ -659,8 +659,9 @@ class jhtree_decl CKeyLevelManager : implements IKeyManager, public CInterface
             if (!started)
             {
                 started = true;
-                segs.checkSize(keyedSize, keyName.get());
-                numsegs = segs.segMonitors.length();
+                numsegs = segs.ordinality();
+                if (numsegs)
+                    segs.checkSize(keyedSize, keyName.get());
```

Comment:
@richardkchapman @ghalliday - I made this change (avoid running checkSize() in reset()) because I did not want to serialize over the segment monitors per BLOB request.
So in the BLOB case there are 0 segment monitors and checkSize() would fail (as it checks size against calculated size from seg. monitors).

I think the change is safe, as it only applies to a key manager who have not had segment monitors setup and for blob access only that is not necessary.

What do you think?